vcd/resource_vcd_nsxt_alb_virtual_service_req_rules.go

```diff
@@ -0,0 +1,772 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdAlbVirtualServiceReqRules() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceVcdAlbVirtualServiceReqRulesCreate,
+		ReadContext:   resourceVcdAlbVirtualServiceReqRulesRead,
+		// Update is the same as create and it does not have any additional details like rule IDs
+		// which are important for updates in some cases.
+		UpdateContext: resourceVcdAlbVirtualServiceReqRulesCreate,
+		DeleteContext: resourceVcdAlbVirtualServiceReqRulesDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVcdAlbVirtualServiceHttpPolicyImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"virtual_service_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "NSX-T ALB Virtual Service ID",
+			},
+			"rule": {
+				Type:        schema.TypeList,
+				Optional:    true,
```

Comment:
Documentation states this is _"(Required) One or more"_, however it is optional here.

Reply:
Hm, That is a good query. Technically - I can run it empty and the API does allow it. But this then becomes equivalent to destroying a resource. Perhaps required makes more sense. Changing.